### PR TITLE
feat(preview): Scroll Preview window command and bindings

### PR DESCRIPTION
**Pull request by alexveden**
I'd like to extend #675 with scrolling capabilities.

1. I took telescope's code for preview scrolling. Code simply sends <C-Y> and <C-E>, nothing complex.
2. Scrolling is performed without focusing the preview window (for me, it's important)
3. Scroll step is configurable

### CODE REVIEW

**Comment by pysan3** [comment]
Thanks for your PR!

This is really interesting and tbh I didn't imagine this feature could be implemented this easily. 

However I believe a lot of users already use `<C-d>, <C-u>` to scroll (jump half page) the neo-tree buffer itself and I think we should prioritize that over scrolling the preview window. 

I would suggest implementing a fallback when `if not Preview:is_active()` just scroll neo-tree window (so don't **always** hijack the keybind), OR, comment out your suggested keymap in defaults.lua by default and let the user choose whether to add it and which key to map to. (For example IMO I'll map it to `<C-f>, <C-b>` cuz I rarely use them to scroll neo-tree). 
What do you think?

FYI you should be able to use `state.winid` but be careful that this valve may be nil.

File: lua/neo-tree/sources/common/preview.lua [hunk 1]
```diff
@@ -435,4 +435,17 @@ Preview.focus = function()
   end
 end
 
+Preview.scroll = function(state)
+  local direction = state.config.direction
+  local input = direction < 0 and [[]] or [[]]
```
[the remark this answers is not in the document]
Comment (alexveden) [reply lua/neo-tree/sources/common/preview.lua:440]: Actually, it's a copy/paste of Telescope function: 

Those symbols are raw ^Y and ^E escape characters, it's GH display issue I believe.

https://github.com/nvim-telescope/telescope.nvim/blob/master/lua/telescope/previewers/buffer_previewer.lua#L310

**Comment by alexveden** [comment]
> Thanks for your PR!

> I would suggest implementing a fallback when `if not Preview:is_active()` to scroll neo-tree window (so don't **always** hijack the keybind), OR, comment out your suggested keymap in defaults.lua by default and let the user choose whether to add it and which key to map to. (For example IMO I'll map it to `<C-f>, <C-b>` cuz I rarely use them to scroll neo-tree). What do you think?
> 
I don't mind about `<C-f> / <C-b>` mappings, actually.  I don't use `<C-d>/<C-u>` neither, just saw people were requesting this in the #675.


>  comment out your suggested keymap in defaults.lua by default and let the user choose whether to add it and which key to map to. 

I would stick to alternative mapping then. I'll push changes tommorrow.

File: lua/neo-tree/sources/common/preview.lua (hunk 1 above)
[the remark this answers is not in the document]
Comment (pysan3) [reply lua/neo-tree/sources/common/preview.lua:440]: I see. Thanks. Could you add a comment there?

**Comment by pysan3** [comment]
> >  comment out your suggested keymap in defaults.lua by default and let the user choose whether to add it and which key to map to. 
> 
> 
> 
> I would stick to alternative mapping then. I'll push changes tommorrow.
> 
> 

Awesome. Please notify that you added the feature in the regarding issues as well so that more people can notice the new feature ;)

And please ping me again after you added the change (my notifications gets flooded pretty quickly 😅).

**Comment by alexveden** [comment]
@pysan3 I've changed the default bindings. Please take a look.

**Comment by alexveden** [comment]
This pull request also related to #672

**Review by pysan3 — APPROVED** [review]

**Pull request merged**